pkg/params/settings/config.go

```diff
@@ -154,106 +136,26 @@ func ConfigToSettings(logger *zap.SugaredLogger, setting *Settings, config map[s
-		setting.ErrorDetection = errorDetection
-	}
-
-	errorDetectNumberOfLines, _ := strconv.Atoi(config[ErrorDetectionNumberOfLinesKey])
-	if setting.ErrorDetection && setting.ErrorDetectionNumberOfLines != errorDetectNumberOfLines {
-		logger.Infof("CONFIG: setting error detection limit of container log to %v", errorDetectNumberOfLines)
-		setting.ErrorDetectionNumberOfLines = errorDetectNumberOfLines
-	}
-
-	if setting.ErrorDetection && setting.ErrorDetectionSimpleRegexp != strings.TrimSpace(config[ErrorDetectionSimpleRegexpKey]) {
-		// replace double backslash with single backslash because kube configmap is giving us things double backslashes
-		logger.Infof("CONFIG: setting error detection regexp to %v", strings.TrimSpace(config[ErrorDetectionSimpleRegexpKey]))
-		setting.ErrorDetectionSimpleRegexp = strings.TrimSpace(config[ErrorDetectionSimpleRegexpKey])
-	}
-
-	if setting.CustomConsoleName != config[CustomConsoleNameKey] {
-		logger.Infof("CONFIG: setting custom console name to %v", config[CustomConsoleNameKey])
-		setting.CustomConsoleName = config[CustomConsoleNameKey]
-	}
-
-	if setting.CustomConsoleURL != config[CustomConsoleURLKey] {
-		logger.Infof("CONFIG: setting custom console url to %v", config[CustomConsoleURLKey])
-		setting.CustomConsoleURL = config[CustomConsoleURLKey]
-	}
-
-	if setting.CustomConsolePRdetail != config[CustomConsolePRDetailKey] {
-		logger.Infof("CONFIG: setting custom console pr detail URL to %v", config[CustomConsolePRDetailKey])
-		setting.CustomConsolePRdetail = config[CustomConsolePRDetailKey]
-	}
-
-	if setting.CustomConsoleNamespaceURL != config[CustomConsoleNamespaceURLKey] {
-		logger.Infof("CONFIG: setting custom console namespace URL to %v", config[CustomConsoleNamespaceURLKey])
-		setting.CustomConsoleNamespaceURL = config[CustomConsoleNamespaceURLKey]
-	}
+	return nil
+}
 
-	if setting.CustomConsolePRTaskLog != config[CustomConsolePRTaskLogKey] {
-		logger.Infof("CONFIG: setting custom console pr task log URL to %v", config[CustomConsolePRTaskLogKey])
-		setting.CustomConsolePRTaskLog = config[CustomConsolePRTaskLogKey]
+func isValidURL(rawURL string) error {
+	if _, err := url.ParseRequestURI(rawURL); err != nil {
+		return fmt.Errorf("invalid value for URL, error: %w", err)
 	}
+	return nil
+}
 
-	rememberOKToTest := StringToBool(config[RememberOKToTestKey])
-	if setting.RememberOKToTest != rememberOKToTest {
-		logger.Infof("CONFIG: setting remember ok-to-test to %v", rememberOKToTest)
-		setting.RememberOKToTest = rememberOKToTest
+func isValidRegex(regex string) error {
+	if _, err := regexp.Compile(regex); err != nil {
+		return fmt.Errorf("invalid regex: %w", err)
 	}
-
 	return nil
 }
 
-func StringToBool(s string) bool {
-	if strings.ToLower(s) == "true" ||
-		strings.ToLower(s) == "yes" || s == "1" {
-		return true
+func startWithHTTPorHTTPS(url string) error {
+	if !strings.HasPrefix(url, "http://") && !strings.HasPrefix(url, "https://") {
```

Comment:
@chmouel 
i have updated this condition a bit
earlier it was
https://github.com/openshift-pipelines/pipelines-as-code/blob/327475944b406e3f2163179592d5e3c6ce6f7590/pkg/params/settings/validation.go#L76
not sure why `strings.HasPrefix(v, "http://")` without a `!` 🤔

Reply:
Bdw the pr is good for review :)

Reply:
nice will review tomorrow, i am not so sure why i made that http thing but will try to dig into it lol ( was it me?)

Reply:
maybe i wanted to enforce only https url but my comment says otherwise, anyway good fix!